compare-script.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# This script computes a comparison among two different versions of the same
+# demo-things repository. Choose the device name you want to compare as input.
+
+# Stops at the first error
+set -e
+
+# Check whether a device name has been passed as input
+if [ -z "$1" ]
+then
+    echo "No device name has been passed as input"
+    exit 1
+fi
+
+# Device name
+DEVICE=$1
+
+# Time filename
+TIME="time.txt"
+# Filesize filename
+FILESIZE="filesize.txt"
+# Snippets directory
+SNIPPETS="snippets"
+# Metrics directory
+METRICS="metrics"
+
+# Comparison directory name
+COMPARE_DIR="Compare"
+
+# First commit sha
+FIRST_COMMIT="ccd8fff"
+# Second commit sha
+SECOND_COMMIT="master"
+
+# Obtains metrics
+function obtain-metrics () {
+    # Compare directory path
+    COMPARE_PATH=$COMPARE_DIR/$1
+
+    # Checkout commit
+    git checkout $1
+
+    # Create version commit directory
+    mkdir -p $COMPARE_PATH
+
+    # Get building time through `/usr/bin/time` tool
+    /usr/bin/time -o $COMPARE_PATH/$TIME cargo build --release --bin $DEVICE
```

Comment:
a `cargo clean` is needed before this, otherwise the second cargo build would use the cached shared deps.